### ISSUE TITLE
PageHeader - Add button properties to Figma component

### ISSUE THREAD

**Issue by vavalos5** [grommet/hpe-design-system#3290]
## Background
During a call with the GLCP team, it came to our attention that the PageHeader component isn't working for teams in terms of having to hide the buttons within the layer panel.

In addition, designers feel that showing all 3 buttons is encouraging to have to use all 3.

To solve this, we can add button properties to hide and add buttons rather than having to hide them in the layer panel.

## Todo's

- [x] Collect use cases from across services ( Platform - Sylvia Shen, <s>CoM - Greg Furuiye</s>, Justin Barton, GLC - Chandan, Bhim)
- [x] Understand what the 80% use case for designs is when using the PageHeader (should we design to have 1-2 buttons, if so, which ones?)
- [x] Insert use cases within this [Figma file](https://www.figma.com/file/BHp0KrOcPvbnzNyshfHkd8/HPE-Page-Header?type=design&node-id=2%3A3&t=P08wM9zSP2ZsX0Az-1). You'll also find previous use cases here that could be helpful for you to understand the needs.
- [x] Share with team reasoning/direction
- [ ] Create a branch in Figma (ask Vicky for help on this)
- [ ] Add properties to Buttons to turn on and off buttons
- [ ] Default PageHeader should be determined by the 80% use case
- [ ] Get a review from Vicky on properties
- [ ] Publish and announce

**Comment by halocline** [comment]
> The default variant when a user pulls in a pageheader should be the 80/20 use case. For pageheader, I'd assume the default variant should just show one primary button.

Maybe one secondary button as the default so that we aren't assuming what the primary action of the page is?

**Comment by vavalos5** [comment]
Please create a branch when working on changes. Let me know if you need help on how to do that.

**Comment by ashifalinadaf** [comment]
I will familiarise with the ticket issue and understand the requirement.  
I will be collecting the use cases from across services.

**Comment by ashifalinadaf** [comment]
I was able to get the use cases for GLCP team.
Collecting the other cases from different teams you mentioned,
I have also added the button properties of page header for ANT, Atlassian and Polaris to understand their views on the same.

**Comment by SylviaShenCC** [comment]
Hi @ashifalinadaf , 
We have very few instances of page-level actions that consist of three buttons and our screen designs almost do not include mockups for smaller screens. Although I don't see that the page component does not require significant updates, we have given limited attention to optimizing the user experience for smaller resolutions. Generally, GLCP users tend to be task-oriented and operate primarily through a monitor. However, services such as marketing may provide valuable insights on how to enhance the button experience on mobile devices. 

Here is the use case that i collected for more complex page level actions https://www.figma.com/file/fNJhhmEO66r8pfFuBEp0Dk/GLCP-Component-research?type=design&node-id=1609%3A98788&t=ECdTnrpWhoRu1DeY-1

**Comment by MikeKingdom** [comment]
> Hi @ashifalinadaf , We have very few instances of page-level actions that consist of three buttons and our screen designs almost do not include mockups for smaller screens. Although I don't see that the page component require significant updates, we have given limited attention to optimizing the user experience for smaller resolutions. Generally, GLCP users tend to be task-oriented and operate primarily through a monitor. However, services such as market place may provide valuable insights on how to enhance the button experience on mobile devices.

@SylviaShenCC note that the contextual help panel being added to GLCP has triggered the lack of smaller screen support an issue across GLCP. When the panel is open the rest of the app is forced into a smaller size which approaches mobile sizes.

**Comment by vavalos5** [comment]
@ashifalinadaf were you able to connect with Greg and collect use cases from him?

**Comment by ashifalinadaf** [comment]
No, @vavalos5 but I did connect with Sylvia and Srinivas (from Storage team) and collected necessary use cases and inputs.

**Comment by ashifalinadaf** [comment]
@SOjaHPE Please review the analysis done on the different use cases and let me know about the next steps or any corrections needed. 
[Figma file](https://www.figma.com/file/BHp0KrOcPvbnzNyshfHkd8/HPE-Page-Header?type=design&node-id=2%3A3&t=P08wM9zSP2ZsX0Az-1).

**Comment by ashifalinadaf** [comment]
As per Scott's guidance, I was able to get the inputs for the final outcomes.
Had a discussion on the same with Scott and doubts were clarified. 
Summarising the findings and will be put for discussion for the team.

**Comment by ashifalinadaf** [comment]
I have gathered some of the button groups from the different services having 80%+ use cases. 
Please review them and let me know if we can use the same for further steps. @SOjaHPE

**Comment by ashifalinadaf** [comment]
Had a discussion with Scott and review comments were discussed and moving forward with next step of creating the button group component - Add properties to Buttons to turn on and off buttons.

**Comment by ashifalinadaf** [comment]
Hello @vavalos5 I have created some button properties for page header based on the research, if you can have a look and comment on the same, will help me to move for the next step. 
[Figma file](https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/branch/Yt6oztChgecuWLR7MhsQ35/HPE-Design-System-Components?type=design&node-id=2079%3A44189&t=l2c7qvSG6hqMkxJG-1) (created a branch) and other [Figma file](https://www.figma.com/file/BHp0KrOcPvbnzNyshfHkd8/HPE-Page-Header?type=design&node-id=2%3A3&t=P08wM9zSP2ZsX0Az-1). (research).

**Comment by vavalos5** [comment]
Great work Asif! 

It's really nice seeing the built out components based on the different use cases. I think as a next step, we have to think about the pros and cons of using slots which we'd then have to deprecate the current PageHeader component and refactor the new one using the the components you created. This would mean that everywhere that this component is being used across services would have to be updated by the designers.

OR we can keep our current component and just add a property where you can swap all of the different variations you've created. This would just update designers that there are new variants available and all they need to do is hit a n "update component" button. this wouldn't affect designer's files and wouldn't have to update unless they want to use one of the new variants.

**Comment by ashifalinadaf** [comment]
- I'd suggest to go with second option - adding a property where we can swap all of the different variations. This property can be brought at the top to use it often. This would help us to not meddle with current page header component and also help the designers to have a choice. 

- Using the slot component can help the designers to quickly swap the button components but this needs effort for us to manage and to educate designers on how to use it. This will also create more components to share and maintain. Components might be swapped with every component in our design system.

**Comment by ashifalinadaf** [comment]
Created button properties with switch on off the button component, submitting for review. [Figma file](https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/branch/Yt6oztChgecuWLR7MhsQ35/HPE-Design-System-Components?type=design&node-id=2079%3A44189&t=l2c7qvSG6hqMkxJG-1)

**Comment by vavalos5** [comment]
Nice work Asif, i've left some comments on your Figma file. 

Inviting @KennyAtHPE to review as well.

**Comment by ashifalinadaf** [comment]
Changes have been done to the file and on/off toggles have been added to page header component.

**Comment by vavalos5** [comment]
Met with Asif. He will be starting a new branch since the other branch he's currently working on has been altered with and is no longer showing our current PAgeHeader component. The current page header component is needed to add the boolean off/on properties to the buttons themselves. In addition, the menu button and the kebab buttons will be added to the set of buttons container so it is available to toggle off/on.

**Comment by SOjaHPE** [comment]
[celebrate]     Oja, Scott reacted to your message:
________________________________

**Comment by ashifalinadaf** [comment]
Thank you Vicky, for looking into and for the comments. 

1. Reflecting one use case on the PageHeader.
I'd suggest to use 'Primary and Secondary Button' for the XL, L and M sizes. 
After the research on various use cases 'Primary button' alone is used in most cases only before the 'Primary and Secondary button', however the DS guidance says 'Most pages wont have Primary Button' hence suggesting to use the 'Primary and Secondary Button' for XL, L and M sizes. 

And also 'Action Menu and Primary Button' for Small size, & 
More icon (action menu icon) with Primary button for xtra small 
(if we can only use the different component restricting 2 buttons for small size and 1 icon and one primary button for Xtra small) had a discussion with @SeamusLeonardHPE , he is helping to find a solution on this. 

But If we have a look at the DS guidance xtra small size shifting itself below the subtitle and aligning to the left. 
<img width="656" alt="Screenshot 2023-06-06 at 4 54 13 PM" src="https://github.com/grommet/hpe-design-system/assets/132882230/d8cf8d54-a5fe-4e15-9c7b-d00f61165b04">
We will have to device different component for XS cases, it wasn't the case for current page header. 

I will be adding the comment and link for the designers to follow the guidance and use the buttons accordingly. 

Working on the 'testframe' and share with others to get the feedback across services.

**Comment by ashifalinadaf** [comment]
Tagged Greg, Justin, Bhim, Sylvia and Srinivas in Figma file for testing the component.

**Comment by ashifalinadaf** [comment]
I had a look at the DS guidance and it says 

'If the screen breakpoint is small or x-small:

If there is a primary action, then all actions should appear beneath the title and subtitle and be left-aligned with page content. To implement, apply responsive={true} to PageHeader.
If there is no primary action, then the overflow menu should remain to the right of the title.'

I have added additional components for Small and X-Small as per DS Guidance. [Figma](https://www.figma.com/file/THoYs4B75xSaahj4uqw6tt?node-id=2079:44189#465372310)

@vavalos5 let me know if this is okay or we can just have x-small component with stacked component as suggested by Chris.

**Comment by ashifalinadaf** [comment]
Discussed with Vicky on the small and x-small component and guidance for the use of page header component. 
X-small size - lower resolution component with stacked button added to the Page header, 
Based on the inputs from @taysea added stacked component to the small size too. 
Tagged designers (Greg, Jeff, Bhim, Sylvia and Srinivas) and developers @taysea and @halocline for the inputs to the page header component. 

And also had a discussion with @KennyAtHPE on boolean properties issue, it was clarified that 2 buttons can be used for all the components and suggested to create another ticket to add the guidance as per the developers input.

**Comment by ashifalinadaf** [comment]
Removing the x-small instance from the Page header component, x-small example of stacked example added. 

I did get some inputs from GLCS team on the test component: 

- It was easy for the designers to toggle between the buttons, time saving. 
- Designer may need to go though the guidance to understand how to use the buttons (number, order, size).
- Concern about the order of the button for action menu and primary button/secondary button this is covered in ticket #3387 

Other inputs will be added from developers and other designers once received.

**Comment by vavalos5** [comment]
Final steps are:

- Get a review from @KennyAtHPE too on guidance and components
- Get a review from @taysea on guidance and component
- Insert the max, gap and other informational visuals as discussed during the call on the examples to the right
- Share out with team during show and tell tomorrow

Once the above has been completed and all is agreed upon, we can transfer them to the main Design System file - connect w/ Vicky on this.

**Comment by vavalos5** [comment]
There was a good question that came in from Sylvia, i'll include below:

"Thanks, Ash. I've checked the page header component and played around with it. It works great. Now that we have the ellipsis in property control pannel, do we have any guidelines to communicate that ellipsis represents a variant for the menu button on small resolutions? Please correct me if my statement is incorrect. My concern is that some designers prefer ellipsis for collapsing buttons, so if we provide them with the option, there might be a misuse where ellipsis is applied on an extra-large resolution. Let me know your thoughts on this."

While this is not a part of this ticket scope, I think Sylvia's question is valid. I don't think we need to restrict designers or be prescriptive about being able to use the actions menu button only at larger resolutions and not the ellipsis. I've seen this in many cases, including here on GitHub. While we don't have an actual "more" icon on GitHub, the Avatar and other icons serve as "more" icons. Any thoughts on this? @halocline @SOjaHPE

**Comment by KennyAtHPE** [comment]
Everything is looking good, I left a few comments on the component and guidance.

**Comment by ashifalinadaf** [comment]
Made changes to the Page header component suggested by the designers.

**Comment by ashifalinadaf** [comment]
I will be making changes to the component to add small/x-small stacked variants, and check with any design issues.

**Comment by ashifalinadaf** [comment]
Changes have been made and checked with minor errors. Up for review. 
@vavalos5 Can you please have a look and let me know. 
[Figma](https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/branch/THoYs4B75xSaahj4uqw6tt/HPE-Design-System-Components?type=design&node-id=2079%3A44189&t=0AjmEqtZRpb0DjMa-1)

**Comment by vavalos5** [comment]
Thanks Asif, i've left some comments on your file. Honestly am wondering whether we need to continue this work if @SeamusLeonardHPE is working on something so similar that is likely going to change the design. 

@halocline @SOjaHPE What do you think?

**Comment by taysea** [comment]
Blocked by #3387

**Comment by SeamusLeonardHPE** [comment]
Resurfaced do to our branch cleaning projects.
https://hpe-my.sharepoint.com/:x:/p/kenny_tran/EagZuftPiF9Hj2m-hR6hiMMBx1_aL8mrD9ZscMFTtQhAVQ?e=ghXMmX

Figma branch linked:

Archived branch to facilitate tidy up.
https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/branch/THoYs4B75xSaahj4uqw6tt/HPE-Design-System-Components?type=design&node-id=2079%3A44189&mode=design&t=FLBe5rPM6G6ddQQ4-1


Branch can be access but should not be merged to main

**Comment by SeamusLeonardHPE** [comment]
Work has been continued on this figma file as part of #3832 

https://www.figma.com/file/05eEXKWnpX8QvNjs3ZxS6b/%233832-PageHeader---Subtitle-width?type=design&node-id=213-4114&mode=design&t=pqH9qh0pnYChvlq2-4

**Comment by vavalos5** [comment]
Any updates on this ticket @SeamusLeonardHPE @ashifalinadaf ? I see it's blocked. Can we specify here so @halocline @SOjaHPE and @SeamusLeonardHPE can help unblock you. Thanks!

**Comment by ashifalinadaf** [comment]
@vavalos5 this was blocked by the other ticket that
@SeamusLeonardHPE is working on button order and menu refactor

**Comment by SeamusLeonardHPE** [comment]
Ive moved this into blocked and will update the labels as its an internal dependancy.
There is a new ticket which I'l resolve this sprint for the [menu component ](https://github.com/grommet/hpe-design-system/issues/3898)
It's the current menu components behaviour, on open, breaks the page header. As soon as thats resolved an it's inserted into the updated page header I'll request a review.
